internal/pkg/api/server/buffer/service/service.go

```diff
@@ -75,6 +60,13 @@ func (*service) CreateReceiver(d dependencies.ForProjectRequest, payload *buffer
 	// Persist receiver
 	err = store.CreateReceiver(ctx, receiver)
 	if err != nil {
+		if errors.Is(err, &dependencies.ReceiverLimitReached{}) {
```

Comment:
Toto je dost nestandardne pouzitie, `errors.Is`, aj ked funkcne:
https://goplay.tools/snippet/9eCnB0lBlYb

Toto je standardne pouzitie, ze je nejaka globalna hodnota, ktora zodpoveda chybe:
https://goplay.tools/snippet/9eCnB0lBlYb

Pozri priklad v docs:
- https://pkg.go.dev/errors#example-Is
- https://pkg.go.dev/io/fs#pkg-variables

-------------------

Takze najprv si treba povedat, ci bude chybova hlaska staticka, a potom `var ... ` + `errors.Is`.

Alebo ci to chces riesit podla typu, a potom `type .... ` + errors.As`.

V kode vratis chybu ako hodnotu, nie pointer.

V  `errors.As` je potom pointer, kam sa ma chyba ulozit, ak zodpoveda.
Kontroluju sa aj vsetky "pod-chyby", ktore moze vratit `Unwrap`.

Pozri priklad na `errors.As`: 
- https://pkg.go.dev/errors#example-Is

-----

Ak chces tak si mozeme k tomu zavolat.
Prakticky to staci spravit tak ako `TemplateNotFoundError`.

Reply:
Ty playground linky cos poslal vedou na stejné id

Reply:
Nechápu. Je to použité stejně, jen tam místo `As` je `Is`, protože nepotřebuju z toho erroru vytáhnout hodnotu

Reply:
Link som opravil.

Reply:
> Nechápu. 

TLDR, nikto to takto nepouziva, je to potrebne vyriesit standardne, aby sa nad tym clovek nemusel pozastavovat, ked cita kod.

Reply:
Takže stačí, když to změním na `As`, i když nepoužiju tu hodnotu (stejně jako s TemplateNotFoundError)? Proč ti to přijde o tolik lepší?

Reply:
Zavolame si? Lebo toto ti funguje iba v tomto konkretnom pripade, v dalsich nie.

Reply:
Mozem ti to vysvetlit podrobnejsie:
https://meet.google.com/dtc-whmo-uou

Reply:
For posterity: `Is` kontroluje reference equality místo typu, takže tohle by přestalo fungovat pokud by `ReceiverLimitReachedError` nebyl empty struct.